extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/CDIDelegatingTransactionManager.java

```diff
@@ -0,0 +1,176 @@
+package io.quarkus.narayana.jta.runtime;
+
+import java.io.*;
+
+import javax.enterprise.context.*;
+import javax.enterprise.event.*;
+import javax.enterprise.inject.*;
+import javax.inject.*;
+import javax.transaction.*;
+
+/**
+ * A delegating transaction manager which receives an instance of Narayana transaction manager
+ * and delegates all calls to it.
+ * On top of it the implementation adds the CDI events processing for {@link TransactionScoped}.
+ */
+@Singleton
+public class CDIDelegatingTransactionManager implements TransactionManager, Serializable {
+    private static final long serialVersionUID = 1598L;
```

Comment:
@mkouba can you even serialize a singleton? I think you rather want app scoped for that, don't you?

Reply:
You can serialize any bean. However, we don't support passivating scopes, serializable client proxies, etc.